proto/cosmos/staking/v1beta1/staking.proto

```diff
@@ -320,6 +320,14 @@ message Params {
     (amino.dont_omitempty) = true,
     (cosmos_proto.scalar)  = "cosmos.Dec"
   ];
+  // key_rotation_fee is fee to be spent when rotating validator's key
+  // (either consensus pubkey or operator key)
+  cosmos.base.v1beta1.Coin key_rotation_fee = 7 [
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+  // max_cons_pubkey_rotations is maximum number of consensus pubkey rotations
+  uint64 max_cons_pubkey_rotations = 8;
```

Comment:
Why is there a max rotations? How is this applied? Per validator?

Reply:
Yeah, this is applied as per the validator, these changes came in from upstream work. let me know if this has to be removed or change to something else.

Reply:
Reading the adr this is suppose to be per unbonding period. We should adjust the code to adhere to the adr.

Reply:
> Why is there a max rotations?

Hey @alexanderbez, this is mentioned in the ADR 16.

Reply:
https://docs.cosmos.network/main/architecture/adr-016-validator-consensus-key-rotation

Reply:
Bummer. Seems I approved that but the design choice doesn't make sense to me.

Reply:
what is your design change, we should double check the adr since it was written a while ago.

Reply:
Given that we need to change slashing signing info and possibly other data structures anyway, I think this is fine.